grafast/website/grafast/step-classes.md

```diff
@@ -159,6 +159,54 @@ console.log("$a = " + $a.toString());
 You may override this to add additional data to the `toString` method (the data
 that would occur between the triangular brackets).
 
+## Access methods
+
+When writing a step class, you should implement either `.at()` or `.get()` depending on
+if the step represents a list, array, or an objet.
+
+### at
+
+Implement `.at()` if your step represents a list or an array. It should accept a single argument, an
+integer, which represents the index within the list-like value which should be accessed.
+
+Usage:
+
+```ts
+const $abTuple = list([$a, $b]);
+
+// abTuple.at(0) returns $a
+```
+
+### get
+
+Implement `.get()` if your step represents an object. It should accept a single argument, a
+string, which represents an attribute to access an object-like value.
```

Comment:
````suggestion
string, which represents an attribute to access an object-like value.

```ts
import { access } from 'grafast';

class MyObjectStep extends ExecutableStep {
  // ...

  get(key) {
    // Your step may implement a more optimized solution here.
    return access(this, key);
  }
}
```
````